Unable to successfully run tests using Python 3.8

I'm attempting to test this project on all supported Python versions, which appears to include Python 3.8 to 3.12. However, I'm getting errors testing on Python 3.8. I believe that this is related to [type hinting syntax that is not supported in 3.8](https://stackoverflow.com/a/63460173).

Here's the error that I'm seeing:

    quantile_forest/tests/test_quantile_forest.py:44: in <module>
    FOREST_REGRESSORS: dict[str, Any] = {
    E   TypeError: 'type' object is not subscriptable

If it's helpful, I'm seeing this error during an automated testing run in a fork of this project. The exact error message and environment I'm using can be seen in the [Actions log](https://github.com/jncraton/quantile-forest/actions/runs/6618064069/job/17975703965).

Is Python 3.8 still supported? Are the tests expected to run correctly for that version?

This issue is related to the [JOSS review](https://github.com/openjournals/joss-reviews/issues/5976) of this project.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -4,15 +4,19 @@ on: [push, pull_request]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+    runs-on: ${{ matrix.os }}
 
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up Python 3.12
+      - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v4
         with:
-          python-version: '3.12'
+          python-version: ${{ matrix.python-version }}
           cache: 'pip'
           cache-dependency-path: pyproject.toml
 

--- a/quantile_forest/tests/test_quantile_forest.py
+++ b/quantile_forest/tests/test_quantile_forest.py
@@ -4,7 +4,7 @@ Testing for the quantile forest module (quantile_forest._quantile_forest).
 
 import math
 import warnings
-from typing import Any
+from typing import Any, Dict
 
 import numpy as np
 import pytest
@@ -41,7 +41,7 @@ perm = rng.permutation(min(california.target.size, 500))
 X_california = california.data[perm]
 y_california = california.target[perm]
 
-FOREST_REGRESSORS: dict[str, Any] = {
+FOREST_REGRESSORS: Dict[str, Any] = {
     "ExtraTreesQuantileRegressor": ExtraTreesQuantileRegressor,
     "RandomForestQuantileRegressor": RandomForestQuantileRegressor,
 }

